_posts/2022-07-01-argocd-image-updater-for-functions.md

```diff
@@ -0,0 +1,248 @@
+---
+title: "Automatically update OpenFaaS functions with the ArgoCD Image Updater"
+description: "We show you how to keep your functions up to date with the Argo CD Image Updater"
+date: 2022-07-01
+categories:
+- gitops
+- argocd
+- functions
+- kubernetes
+author_staff_member: han
+---
+
+We show you how to keep your functions up to date with the Argo CD Image Updater
+
+## Tutorial
+In this tutorial we are going to use [Argo CD](https://argoproj.github.io/cd/) to bootstrap a new cluster. We will demonstrate how you can use the [app-of-apps pattern](https://argo-cd.readthedocs.io/en/stable/operator-manual/cluster-bootstrapping/) to declaratively manage both OpenFaaS and a set of OpenFaaS functions. We will then deploy and configure the [Argo CD Image Updater](https://argocd-image-updater.readthedocs.io/en/stable/) to automatically update the function images to their latest allowed version.
+
+### How it works
+
+> Include diagram.
+
+Argo CD is a declarative, GitOps continuous delivery tool for Kubernetes. It allows you to use Git repositories as the source of truth for defining the desired application state.
+
+Argo CD automates the deployment of the desired application states in the specified target environments. It will continuously monitor the running applications and compare the current live state against the desired state. It reports any deviations and depending on your configuration automatically updates the application to the desired state.
+
+The **app of apps pattern** is commonly used to install multiple applications in a cluster. Its concept is to declaratively specify one Argo CD app that only consists of other apps. In our case we want to create and deploy two Argo CD apps, one for the OpenFaaS Operator and one for our functions. We will need to create an app called applications that consists of both the openfaas-operator and openfaas-functions app.
+
+The Argo CD Image updater is used to automatically update functions when a new version is published. It polls the image registry to check if a new version of an image is found. If a new version is available and the version constraint is met, the Image Updater instructs Argo CD to update the application with the new image.
+
+![app-of-apps](/images/2022-07-argocd-image-updater/app-of-apps.png)
+> Pictured: The Argo CD UI showing the `applications` app containing both the `openfaas-operator` and `openfaas-functions` app.
+
+### Prerequisites
+
+[arkade](https://github.com/alexellis/arkade) lets developers install the latest versions of their favourite tools and Kubernetes apps. We will use it in this tutorial to install Argo CD and get some other required tools.
```

Comment:
`lets developers install` => `enables developers to install`

Reply:
@rgee could "means" also work? "arkade means developers get to install the.."

Reply:
Went for @rgee0 proposal here